### PR TITLE
Progress: implement IPC for windows

### LINKED ISSUE #20105: implement std.Progress IPC for Windows
Extracted from #20059.

Currently, on Windows, `std.Progress` is implemented for the root process that owns the terminal, however it is not able to collect progress information from the child processes that it spawns.

This issue can be closed when child processes using `std.Progress` on Windows communicate their progress to parent processes.

A simple way to test this would be adapting this example to also work on Windows:

https://gist.github.com/andrewrk/b22b4f663cef6b4672d7097de95ea343

For POSIX systems this is implemented by a dedicated pipe file descriptor that is passed from parent to child, much like stdio streams. It should be possible to do something equivalent on Windows.

### PR DESCRIPTION
**Pull request by jacobly0**
The nearby local maximum API surface that would have been required to implement this feature would have introduced a bunch of useless new kernel32 dependencies, so I instead fled two undocumented layers deep into ntdll for the identical functionality in a single syscall.

Closes #20105

### CODE REVIEW

File: lib/std/os/windows/kernel32.zig [hunk 1]
```diff
@@ -464,3 +466,7 @@ pub extern "kernel32" fn RegOpenKeyExW(
 ) callconv(WINAPI) LSTATUS;
 
 pub extern "kernel32" fn GetPhysicallyInstalledSystemMemory(TotalMemoryInKilobytes: *ULONGLONG) BOOL;
+
+pub extern "kernel32" fn IsDebuggerPresent() BOOL;
```
Comment (The-King-of-Toasters) [inline lib/std/os/windows/kernel32.zig:470]: I don't believe these are used anywhere.

File: lib/std/process/Child.zig [hunk 2]
```diff
@@ -1270,61 +1219,148 @@ fn windowsCreateProcessPathExt(
...
+    var handle_list_len: usize = 0;
+    inline for (StdHandles.fields) |field| {
+        if (handles.get(field)) |handle| {
+            if (mem.indexOfScalar(windows.HANDLE, handle_list[0..handle_list_len], handle) == null) {
+                handle_list[handle_list_len] = handle;
+                handle_list_len += 1;
+            }
+        }
+    }
+    switch (windows.ntdll.NtCreateUserProcess(
```
Comment (The-King-of-Toasters) [inline lib/std/process/Child.zig:1281]: Use `NtCreateUserProcess` at your own peril. Relevant quote from @zodiacon's "Windows Native API Programming":

> There have been several attempts in the Infosec community to utilize `RtlCreateUserProcess` and/or `NtCreateUserProcess` to allow running Windows subsystem applications, with varying degrees of success. Part of the problem is the need to communicate with the Windows Subsystem process (`csrss.exe`) to notify it of the new process and thread. This turns out to be fragile, as different Windows versions may have somewhat different expectations.

Essentially: you're one the hook if this breaks on newer Windows versions and/or Wine updates.

**Review by The-King-of-Toasters — CHANGES_REQUESTED** [review]
This PR is great, but it goes much farther than just implementing progress IPC for Windows: You've basically rewritten all of the OS API bindings in the process. To be clear: that's a good thing! But this PR needs to be split up into smaller commits to facilitate better reviews.

Furthermore, I'd like to see some tests that ensure the new `ACCESS_MASK` builder gives the same results as a bitflag before I feel comfortable using this.

Finally, see my comment on `NtCreateUserProcess`.

**Comment by andrewrk** [comment]
2 months old, no updates. I think we have vague plans to try doing this with shared memory instead, anyway.

**Pull request closed without merging**